L1Trigger/TrackFindingTracklet/interface/Settings.h

```diff
@@ -736,7 +736,8 @@ namespace trklet {
     bool doMultipleMatches_{true};
 
     // if true, run a dummy fit, producing TTracks directly from output of tracklet pattern reco stage
-    bool fakefit_{false};
+    // fakefit_=true also required if using TrackerTFP/interface/KalmanFilter.h track fit.
+    bool fakefit_{true};
```

Comment:
We can't set fakefit_=true by default in Settings.h, as it will break the still widely used option of running Hybrid with the old KF. (We need to keep this option until the new KF includes 5-param fits and HO corrections etc). I suggest adding a function setFakeFit(bool x){fakeFit_ = x;} to Settings.h, and calling it from https://github.com/cms-L1TK/cmssw/blob/tschuh/L1Trigger/TrackFindingTracklet/plugins/L1FPGATrackProducer.cc#L257 (where similar "set" functions are called) if a python cfg param "UseNewKF=True". The value of this cfg could be accessed near https://github.com/cms-L1TK/cmssw/blob/tschuh/L1Trigger/TrackFindingTracklet/plugins/L1FPGATrackProducer.cc#L257 (where other cfg params are determined) and it would be set to False near https://github.com/cms-L1TK/cmssw/blob/tschuh/L1Trigger/TrackFindingTracklet/python/Tracklet_cfi.py#L15 . The script TrackFindingTracklet/test/HybridTracksNewKF_cfg.py would override this, setting UseNewKF=True.

Reply:
done

Reply:
We can't set fakefit_=true by default in Settings.h, as it will break the still widely used option of running Hybrid with the old KF. (We need to keep this option until the new KF includes 5-param fits and HO corrections etc). I suggest adding a function setFakeFit(bool x){fakeFit_ = x;} to Settings.h, and calling it from https://github.com/cms-L1TK/cmssw/blob/tschuh/L1Trigger/TrackFindingTracklet/plugins/L1FPGATrackProducer.cc#L257 (where similar "set" functions are called) if a python cfg param "UseNewKF=True". The value of this cfg could be accessed near https://github.com/cms-L1TK/cmssw/blob/tschuh/L1Trigger/TrackFindingTracklet/plugins/L1FPGATrackProducer.cc#L257 (where other cfg params are determined) and it would be set to False near https://github.com/cms-L1TK/cmssw/blob/tschuh/L1Trigger/TrackFindingTracklet/python/Tracklet_cfi.py#L15 . The script TrackFindingTracklet/test/HybridTracksNewKF_cfg.py would override this, setting UseNewKF=True.

Reply:
I think is was done as I wrote it ans it is still done. I guess I solved the problem in a different way as you would solve it?